src/utils/tensor.js

```diff
@@ -36,6 +36,7 @@ const DataTypeMap = new Map([
 
 const ONNXTensor = ONNX.Tensor;
 
+/** @ts-ignore Base constructors must all have the same return type. ts(2510) */
```

Comment:
Oh that's interesting... 👀 In that case, it might be worth making the update to not extend from the base tensor class now, since there are many other typing issues which are showing. See [here](https://github.com/xenova/transformers.js/blob/f8bc912d31b011c5997c253686974338e327ac7b/src/utils/tensor.js#L35) to see what I'm referring to.

Reply:
Nice, you implemented every method that was used from `ONNX.Tensor` into transformers `Tensor` class? I like that, it keeps everything a bit closer to this code and easier to understand.

I guess it may break semver/backwards-compatibility a bit if people on v2 still use ONNX.Tensor methods that a maybe not re-implemented yet? So at least for v3 its a good deal.

Reply:
@xenova Thank you for the info! Currently I'm low on time, so I would appreciate if you can do it.